File: backend/src/models/device.py
"""Device-related Pydantic models."""
from datetime import datetime

from pydantic import BaseModel


class DeviceRegisterRequest(BaseModel):
    """Request model for device registration."""
    mac_address: str
    firmware_version: str | None = None
    sensor_types: list[str] | None = None


class DeviceRegisterResponse(BaseModel):
    """Response model for device registration."""
    device_id: str
    mqtt_username: str
    mqtt_password: str  # Plaintext, returned only on registration
    mqtt_host: str
    mqtt_port: int


class DeviceResponse(BaseModel):
    """Response model for device information."""
    id: str
    mac_address: str
    mqtt_username: str
    plant_id: str | None
    status: str
    firmware_version: str | None
    sensor_types: list[str] | None
    last_seen_at: datetime | None
    created_at: datetime


class DeviceListResponse(BaseModel):
    """Response model for device list."""
    devices: list[DeviceResponse]
    total: int

File: frontend/src/types/device.ts
export interface Device {
  id: string;
  mac_address: string;
  mqtt_username: string;
  plant_id: string | null;
  status: 'provisioning' | 'online' | 'offline' | 'error';
  firmware_version: string | null;
  sensor_types: string[] | null;
  last_seen_at: string | null;
  created_at: string;
}

export interface DeviceListResponse {
  devices: Device[];
  total: number;
}

export interface DeviceRegisterRequest {
  mac_address: string;
  firmware_version?: string;
  sensor_types?: string[];
}

export interface DeviceRegisterResponse {
  device_id: string;
  mqtt_username: string;
  mqtt_password: string;
  mqtt_host: string;
  mqtt_port: number;
}

export interface DeviceProvisionRequest {
  plant_id: string;
}
